syntax = "proto3";

enum MoveIdentifier {
  UNKNOWN_MOVE = 0;
  POUND = 1;
  KARATE_CHOP = 2;
  DOUBLESLAP = 3;
  COMET_PUNCH = 4;
  MEGA_PUNCH = 5;
  PAY_DAY = 6;
  FIRE_PUNCH = 7;
  ICE_PUNCH = 8;
  THUNDERPUNCH = 9;
  SCRATCH = 10;
  VICEGRIP = 11;
  GUILLOTINE = 12;
  RAZOR_WIND = 13;
  SWORDS_DANCE = 14;
  CUT = 15;
  GUST = 16;
  WING_ATTACK = 17;
  WHIRLWIND = 18;
  FLY = 19;
  BIND = 20;
  SLAM = 21;
  VINE_WHIP = 22;
  STOMP = 23;
  DOUBLE_KICK = 24;
  MEGA_KICK = 25;
  JUMP_KICK = 26;
  ROLLING_KICK = 27;
  SAND_ATTACK = 28;
  HEADBUTT = 29;
  HORN_ATTACK = 30;
  FURY_ATTACK = 31;
  HORN_DRILL = 32;
  TACKLE = 33;
  BODY_SLAM = 34;
  WRAP = 35;
  TAKE_DOWN = 36;
  THRASH = 37;
  DOUBLE_EDGE = 38;
  TAIL_WHIP = 39;
  POISON_STING = 40;
  TWINEEDLE = 41;
  PIN_MISSILE = 42;
  LEER = 43;
  BITE = 44;
  GROWL = 45;
  ROAR = 46;
  SING = 47;
  SUPERSONIC = 48;
  SONICBOOM = 49;
  DISABLE = 50;
  ACID = 51;
  EMBER = 52;
  FLAMETHROWER = 53;
  MIST = 54;
  WATER_GUN = 55;
  HYDRO_PUMP = 56;
  SURF = 57;
  ICE_BEAM = 58;
  BLIZZARD = 59;
  PSYBEAM = 60;
  BUBBLEBEAM = 61;
  AURORA_BEAM = 62;
  HYPER_BEAM = 63;
  PECK = 64;
  DRILL_PECK = 65;
  SUBMISSION = 66;
  LOW_KICK = 67;
  COUNTER = 68;
  SEISMIC_TOSS = 69;
  STRENGTH = 70;
  ABSORB = 71;
  MEGA_DRAIN = 72;
  LEECH_SEED = 73;
  GROWTH = 74;
  RAZOR_LEAF = 75;
  SOLARBEAM = 76;
  POISONPOWDER = 77;
  STUN_SPORE = 78;
  SLEEP_POWDER = 79;
  PETAL_DANCE = 80;
  STRING_SHOT = 81;
  DRAGON_RAGE = 82;
  FIRE_SPIN = 83;
  THUNDERSHOCK = 84;
  THUNDERBOLT = 85;
  THUNDER_WAVE = 86;
  THUNDER = 87;
  ROCK_THROW = 88;
  EARTHQUAKE = 89;
  FISSURE = 90;
  DIG = 91;
  TOXIC = 92;
  CONFUSION = 93;
  PSYCHIC_M = 94;
  HYPNOSIS = 95;
  MEDITATE = 96;
  AGILITY = 97;
  QUICK_ATTACK = 98;
  RAGE = 99;
  TELEPORT = 100;
  NIGHT_SHADE = 101;
  MIMIC = 102;
  SCREECH = 103;
  DOUBLE_TEAM = 104;
  RECOVER = 105;
  HARDEN = 106;
  MINIMIZE = 107;
  SMOKESCREEN = 108;
  CONFUSE_RAY = 109;
  WITHDRAW = 110;
  DEFENSE_CURL = 111;
  BARRIER = 112;
  LIGHT_SCREEN = 113;
  HAZE = 114;
  REFLECT = 115;
  FOCUS_ENERGY = 116;
  BIDE = 117;
  METRONOME = 118;
  MIRROR_MOVE = 119;
  SELFDESTRUCT = 120;
  EGG_BOMB = 121;
  LICK = 122;
  SMOG = 123;
  SLUDGE = 124;
  BONE_CLUB = 125;
  FIRE_BLAST = 126;
  WATERFALL = 127;
  CLAMP = 128;
  SWIFT = 129;
  SKULL_BASH = 130;
  SPIKE_CANNON = 131;
  CONSTRICT = 132;
  AMNESIA = 133;
  KINESIS = 134;
  SOFTBOILED = 135;
  HI_JUMP_KICK = 136;
  GLARE = 137;
  DREAM_EATER = 138;
  POISON_GAS = 139;
  BARRAGE = 140;
  LEECH_LIFE = 141;
  LOVELY_KISS = 142;
  SKY_ATTACK = 143;
  TRANSFORM = 144;
  BUBBLE = 145;
  DIZZY_PUNCH = 146;
  SPORE = 147;
  FLASH = 148;
  PSYWAVE = 149;
  SPLASH = 150;
  ACID_ARMOR = 151;
  CRABHAMMER = 152;
  EXPLOSION = 153;
  FURY_SWIPES = 154;
  BONEMERANG = 155;
  REST = 156;
  ROCK_SLIDE = 157;
  HYPER_FANG = 158;
  SHARPEN = 159;
  CONVERSION = 160;
  TRI_ATTACK = 161;
  SUPER_FANG = 162;
  SLASH = 163;
  SUBSTITUTE = 164;
  STRUGGLE = 165;
  SKETCH = 166;
  TRIPLE_KICK = 167;
  THIEF = 168;
  SPIDER_WEB = 169;
  MIND_READER = 170;
  NIGHTMARE = 171;
  FLAME_WHEEL = 172;
  SNORE = 173;
  CURSE = 174;
  FLAIL = 175;
  CONVERSION2 = 176;
  AEROBLAST = 177;
  COTTON_SPORE = 178;
  REVERSAL = 179;
  SPITE = 180;
  POWDER_SNOW = 181;
  PROTECT = 182;
  MACH_PUNCH = 183;
  SCARY_FACE = 184;
  FAINT_ATTACK = 185;
  SWEET_KISS = 186;
  BELLY_DRUM = 187;
  SLUDGE_BOMB = 188;
  MUD_SLAP = 189;
  OCTAZOOKA = 190;
  SPIKES = 191;
  ZAP_CANNON = 192;
  FORESIGHT = 193;
  DESTINY_BOND = 194;
  PERISH_SONG = 195;
  ICY_WIND = 196;
  DETECT = 197;
  BONE_RUSH = 198;
  LOCK_ON = 199;
  OUTRAGE = 200;
  SANDSTORM = 201;
  GIGA_DRAIN = 202;
  ENDURE = 203;
  CHARM = 204;
  ROLLOUT = 205;
  FALSE_SWIPE = 206;
  SWAGGER = 207;
  MILK_DRINK = 208;
  SPARK = 209;
  FURY_CUTTER = 210;
  STEEL_WING = 211;
  MEAN_LOOK = 212;
  ATTRACT = 213;
  SLEEP_TALK = 214;
  HEAL_BELL = 215;
  RETURN = 216;
  PRESENT = 217;
  FRUSTRATION = 218;
  SAFEGUARD = 219;
  PAIN_SPLIT = 220;
  SACRED_FIRE = 221;
  MAGNITUDE = 222;
  DYNAMICPUNCH = 223;
  MEGAHORN = 224;
  DRAGONBREATH = 225;
  BATON_PASS = 226;
  ENCORE = 227;
  PURSUIT = 228;
  RAPID_SPIN = 229;
  SWEET_SCENT = 230;
  IRON_TAIL = 231;
  METAL_CLAW = 232;
  VITAL_THROW = 233;
  MORNING_SUN = 234;
  SYNTHESIS = 235;
  MOONLIGHT = 236;
  HIDDEN_POWER = 237;
  CROSS_CHOP = 238;
  TWISTER = 239;
  RAIN_DANCE = 240;
  SUNNY_DAY = 241;
  CRUNCH = 242;
  MIRROR_COAT = 243;
  PSYCH_UP = 244;
  EXTREMESPEED = 245;
  ANCIENTPOWER = 246;
  SHADOW_BALL = 247;
  FUTURE_SIGHT = 248;
  ROCK_SMASH = 249;
  WHIRLPOOL = 250;
  BEAT_UP = 251;
}

enum PokemonSpecies {
  BULBASAUR = 0;
  IVYSAUR = 1;
  VENUSAUR = 2;
  CHARMANDER = 3;
  CHARMELEON = 4;
  CHARIZARD = 5;
  SQUIRTLE = 6;
  WARTORTLE = 7;
  BLASTOISE = 8;
  CATERPIE = 9;
  METAPOD = 10;
  BUTTERFREE = 11;
  WEEDLE = 12;
  KAKUNA = 13;
  BEEDRILL = 14;
  PIDGEY = 15;
  PIDGEOTTO = 16;
  PIDGEOT = 17;
  RATTATA = 18;
  RATICATE = 19;
  SPEAROW = 20;
  FEAROW = 21;
  EKANS = 22;
  ARBOK = 23;
  PIKACHU = 24;
  RAICHU = 25;
  SANDSHREW = 26;
  SANDSLASH = 27;
  NIDORAN_F = 28;
  NIDORINA = 29;
  NIDOQUEEN = 30;
  NIDORAN_M = 31;
  NIDORINO = 32;
  NIDOKING = 33;
  CLEFAIRY = 34;
  CLEFABLE = 35;
  VULPIX = 36;
  NINETALES = 37;
  JIGGLYPUFF = 38;
  WIGGLYTUFF = 39;
  ZUBAT = 40;
  GOLBAT = 41;
  ODDISH = 42;
  GLOOM = 43;
  VILEPLUME = 44;
  PARAS = 45;
  PARASECT = 46;
  VENONAT = 47;
  VENOMOTH = 48;
  DIGLETT = 49;
  DUGTRIO = 50;
  MEOWTH = 51;
  PERSIAN = 52;
  PSYDUCK = 53;
  GOLDUCK = 54;
  MANKEY = 55;
  PRIMEAPE = 56;
  GROWLITHE = 57;
  ARCANINE = 58;
  POLIWAG = 59;
  POLIWHIRL = 60;
  POLIWRATH = 61;
  ABRA = 62;
  KADABRA = 63;
  ALAKAZAM = 64;
  MACHOP = 65;
  MACHOKE = 66;
  MACHAMP = 67;
  BELLSPROUT = 68;
  WEEPINBELL = 69;
  VICTREEBEL = 70;
  TENTACOOL = 71;
  TENTACRUEL = 72;
  GEODUDE = 73;
  GRAVELER = 74;
  GOLEM = 75;
  PONYTA = 76;
  RAPIDASH = 77;
  SLOWPOKE = 78;
  SLOWBRO = 79;
  MAGNEMITE = 80;
  MAGNETON = 81;
  FARFETCH_D = 82;
  DODUO = 83;
  DODRIO = 84;
  SEEL = 85;
  DEWGONG = 86;
  GRIMER = 87;
  MUK = 88;
  SHELLDER = 89;
  CLOYSTER = 90;
  GASTLY = 91;
  HAUNTER = 92;
  GENGAR = 93;
  ONIX = 94;
  DROWZEE = 95;
  HYPNO = 96;
  KRABBY = 97;
  KINGLER = 98;
  VOLTORB = 99;
  ELECTRODE = 100;
  EXEGGCUTE = 101;
  EXEGGUTOR = 102;
  CUBONE = 103;
  MAROWAK = 104;
  HITMONLEE = 105;
  HITMONCHAN = 106;
  LICKITUNG = 107;
  KOFFING = 108;
  WEEZING = 109;
  RHYHORN = 110;
  RHYDON = 111;
  CHANSEY = 112;
  TANGELA = 113;
  KANGASKHAN = 114;
  HORSEA = 115;
  SEADRA = 116;
  GOLDEEN = 117;
  SEAKING = 118;
  STARYU = 119;
  STARMIE = 120;
  MR_MIME = 121;
  SCYTHER = 122;
  JYNX = 123;
  ELECTABUZZ = 124;
  MAGMAR = 125;
  PINSIR = 126;
  TAUROS = 127;
  MAGIKARP = 128;
  GYARADOS = 129;
  LAPRAS = 130;
  DITTO = 131;
  EEVEE = 132;
  VAPOREON = 133;
  JOLTEON = 134;
  FLAREON = 135;
  PORYGON = 136;
  OMANYTE = 137;
  OMASTAR = 138;
  KABUTO = 139;
  KABUTOPS = 140;
  AERODACTYL = 141;
  SNORLAX = 142;
  ARTICUNO = 143;
  ZAPDOS = 144;
  MOLTRES = 145;
  DRATINI = 146;
  DRAGONAIR = 147;
  DRAGONITE = 148;
  MEWTWO = 149;
  MEW = 150;
  CHIKORITA = 151;
  BAYLEEF = 152;
  MEGANIUM = 153;
  CYNDAQUIL = 154;
  QUILAVA = 155;
  TYPHLOSION = 156;
  TOTODILE = 157;
  CROCONAW = 158;
  FERALIGATR = 159;
  SENTRET = 160;
  FURRET = 161;
  HOOTHOOT = 162;
  NOCTOWL = 163;
  LEDYBA = 164;
  LEDIAN = 165;
  SPINARAK = 166;
  ARIADOS = 167;
  CROBAT = 168;
  CHINCHOU = 169;
  LANTURN = 170;
  PICHU = 171;
  CLEFFA = 172;
  IGGLYBUFF = 173;
  TOGEPI = 174;
  TOGETIC = 175;
  NATU = 176;
  XATU = 177;
  MAREEP = 178;
  FLAAFFY = 179;
  AMPHAROS = 180;
  BELLOSSOM = 181;
  MARILL = 182;
  AZUMARILL = 183;
  SUDOWOODO = 184;
  POLITOED = 185;
  HOPPIP = 186;
  SKIPLOOM = 187;
  JUMPLUFF = 188;
  AIPOM = 189;
  SUNKERN = 190;
  SUNFLORA = 191;
  YANMA = 192;
  WOOPER = 193;
  QUAGSIRE = 194;
  ESPEON = 195;
  UMBREON = 196;
  MURKROW = 197;
  SLOWKING = 198;
  MISDREAVUS = 199;
  UNOWN = 200;
  WOBBUFFET = 201;
  GIRAFARIG = 202;
  PINECO = 203;
  FORRETRESS = 204;
  DUNSPARCE = 205;
  GLIGAR = 206;
  STEELIX = 207;
  SNUBBULL = 208;
  GRANBULL = 209;
  QWILFISH = 210;
  SCIZOR = 211;
  SHUCKLE = 212;
  HERACROSS = 213;
  SNEASEL = 214;
  TEDDIURSA = 215;
  URSARING = 216;
  SLUGMA = 217;
  MAGCARGO = 218;
  SWINUB = 219;
  PILOSWINE = 220;
  CORSOLA = 221;
  REMORAID = 222;
  OCTILLERY = 223;
  DELIBIRD = 224;
  MANTINE = 225;
  SKARMORY = 226;
  HOUNDOUR = 227;
  HOUNDOOM = 228;
  KINGDRA = 229;
  PHANPY = 230;
  DONPHAN = 231;
  PORYGON2 = 232;
  STANTLER = 233;
  SMEARGLE = 234;
  TYROGUE = 235;
  HITMONTOP = 236;
  SMOOCHUM = 237;
  ELEKID = 238;
  MAGBY = 239;
  MILTANK = 240;
  BLISSEY = 241;
  RAIKOU = 242;
  ENTEI = 243;
  SUICUNE = 244;
  LARVITAR = 245;
  PUPITAR = 246;
  TYRANITAR = 247;
  LUGIA = 248;
  HO_OH = 249;
  CELEBI = 250;
}

enum ItemIdentifier {
  UNKNOWN_ITEM = 0;
  FULL_HEAL = 38;
  MAX_POTION = 15;
  FULL_RESTORE = 14;
  POTION = 18;
  SUPER_POTION = 17;
  HYPER_POTION = 16;
  X_ACCURACY = 33;
  GUARD_SPEC = 41;
  DIRE_HIT = 44;
  X_ATTACK = 49;
  X_DEFEND = 51;
  X_SPEED = 52;
  X_SPECIAL = 53;
}